### PR TITLE
Continuous delivery playbook to create an AMI.

### PR DESCRIPTION
**Pull request by macdiesel**
@doctoryes @feanil PTAL.

Please note that playbooks/library/ec2_ami_local.py has been pulled in from ansible 2.0.0.1 to be able to support ami tagging. See #2881 for reference.

Jira: https://openedx.atlassian.net/browse/TE-1250

### CODE REVIEW

**Comment by feanil** [comment]
The AMI also needs to be tagged with all the tags that are on the instance as well the version of configuration and configuration_secure that were used to make it: https://github.com/edx/configuration/blob/master/util/vpc-tools/abbey.py#L543-L566

File: playbooks/library/ec2_ami_local.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
+#!/usr/bin/python
```
Comment (feanil) [inline playbooks/library/ec2_ami_local.py]: This should be named something like `ec2_ami_2_x_x` to indicate the version of ansible it was pulled from.  This makes it easier for us to know which up-stream version it came from and when we can get rid of it.

Comment (macdiesel) [inline playbooks/library/ec2_ami_local.py]: would you prefer the version of ansible this was pulled from or the version of ansible that we needed to be on for the missing feature?

**Comment by macdiesel** [comment]
@feanil @doctoryes comments addressed PTAL. 

I named the ec2_ami module ec2_ami_2_0_0_1 as that was the version I pulled this from and is also the latest current ansible release. This release looks like a bug fix release so the version is both the version it was pulled from as well as the version that if we surpass should remove this module.

File: playbooks/continuous_delivery/create_ami.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  - name: Create AMI
+    ec2_ami_2_0_0_1:
+      instance_id: "{{ instance_id }}"
+      name: "{{ cluster }} -- {{ cluster_version }}"
+      region: "{{ ec2_region }}"
+      wait: "{{ ami_wait }}"
+      wait_timeout: "{{ ami_creation_timeout }}"
+      no_reboot: "{{ no_reboot }}"
+      description: "AMI built via edX continious delivery pipeline - Ansible version: {{ ansible_version }}"
```
Comment (doctoryes) [inline playbooks/continuous_delivery/create_ami.yml]: Nit: typo for `continuous`.

**Comment by doctoryes** [comment]
:+1:

File: playbooks/continuous_delivery/create_ami.yml (hunk 2 above)
Comment (macdiesel) [inline playbooks/continuous_delivery/create_ami.yml]: fixed.

**Comment by macdiesel** [comment]
@feanil this is ready for another look.

File: playbooks/continuous_delivery/create_ami.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      mode: 0600
+
+  - name: Send Hipchat notification AMI is finished baking
+    hipchat:
+      api: "{{ hipchat_url }}"
+      token: "{{ hipchat_token }}"
+      room: "{{ hipchat_room }}"
+      msg: "Finished baking AMI for: {{ play }} \n
+            AMI-ID: {{ ami_register.image_id }} \n
+            {{ play }} repo: {{ cluster_repo }} \n
```
Comment (feanil) [inline playbooks/continuous_delivery/create_ami.yml]: We only need to know the AMI ID. Alton can show you all tags that are currently on an AMI.

**Comment by feanil** [comment]
:sheep: :it:

**Pull request merged**